### ISSUE TITLE
Assignment with command substitution fails as condition

### ISSUE THREAD

**Issue by beltaurus** [koalaman/shellcheck#1685]
#### For bugs
- Rule Id (if any, e.g. SC1000):  syntax error
- My shellcheck version (`shellcheck --version` or 'online'): 0.6.0
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [x] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
if ! _exe=$(command -v dummy); then
  echo "_exe($_exe) does not exist"
  fi
```

#### Here's what shellcheck currently says:

Remote (WSL) session using windows-installation of shellchecker 6.0.0
![image](https://user-images.githubusercontent.com/54114043/64495179-6d524880-d24b-11e9-9a6b-9757975201a5.png)

Note that the perceived syntax problem goes away if I remove the "NOT":
![image](https://user-images.githubusercontent.com/54114043/64495306-35e49b80-d24d-11e9-8d3a-c16843b55352.png)

#### Here's what I wanted or expected to see:
Local session with SAME windows-installation of shellchecker 6.0.0
![image](https://user-images.githubusercontent.com/54114043/64495264-a2ab6600-d24c-11e9-8330-d21ac46a0ebb.png)

**Comment by koalaman** [comment]
ShellCheck itself supports this syntax just fine. Your error appears to be from another tool: https://github.com/tree-sitter/tree-sitter-bash/issues/51
